lib/cocoapods/sources_manager.rb

```diff
@@ -142,41 +137,64 @@ def search_by_name(query, full_text_search = false)
         sets
       end
 
-      # Creates or updates the search data and returns it. The search data
-      # groups by name the following information for each set:
+      # Returns the search data. If a saved search data exists, retrieves it from file and returns it.
+      # Else, creates the search data from scratch, saves it to file system, and returns it.
+      # Search data is grouped by source repos. For each source, it contains a hash where keys are words
+      # and values are the pod names containing corresponding word.
       #
+      # For each source, list of unique words are generated from the following spec information.
       #   - version
       #   - summary
       #   - description
       #   - authors
       #
-      # @note   This operation is fairly expensive, because of the YAML
-      #         conversion.
-      #
-      # @return [Hash{String => String}] The up to date search data.
+      # @return [Hash{String => Hash{String => Array<String>}}] The up to date search data.
       #
       def updated_search_index
+        index = stored_search_index
+        unless index
+          # Create index from scratch
+          UI.puts 'Creating search index..'
+          index = {}
+          all.each do |source|
+            source_name = source.name
+            index[source_name] = aggregate.generate_search_index_for_source(source)
+          end
+          save_search_index(index)
+          UI.puts 'Search index creation done!'
+        end
+        index
+      end
+
+      # Returns the search data stored in the file system.
+      # If existing data in the file system is not valid, returns nil.
+      #
+      def stored_search_index
         unless @updated_search_index
           if search_index_path.exist?
-            require 'yaml'
-            stored_index = YAML.load(search_index_path.read)
-            if stored_index && stored_index.is_a?(Hash)
-              search_index = aggregate.update_search_index(stored_index)
-            else
-              search_index = aggregate.generate_search_index
+            require 'json'
+            index = JSON.parse(search_index_path.read)
+            if index && index.is_a?(Hash) # TODO: should we also check if hash has correct hierarchy?
+              return @updated_search_index = index
             end
-          else
-            search_index = aggregate.generate_search_index
           end
-
-          File.open(search_index_path, 'w') do |file|
-            file.write(search_index.to_yaml)
-          end
-          @updated_search_index = search_index
+          @updated_search_index = nil
         end
         @updated_search_index
       end
 
+      # Stores given search data in the file system.
+      # @param [Hash] index
+      #        Index to be saved in file system
+      #
+      def save_search_index(index)
+        require 'json'
+        @updated_search_index = index
+        File.open(search_index_path, 'w') do |file|
```

Comment:
I think this way works too. Current implementation also performs open in the same way as the following:

```
      File.open(search_index_path, 'w') do |file|
        file.write(search_index.to_yaml)
      end
```

What kind of problem may we face?